ui/app/src/components/RenameAssetDialog/RenameAssetDialogContainer.tsx

```diff
@@ -0,0 +1,238 @@
+/*
+ * Copyright (C) 2007-2022 Crafter Software Corporation. All Rights Reserved.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as published by
+ * the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import { RenameAssetContainerProps } from './utils';
+import { useEnhancedDialogContext } from '../EnhancedDialog';
+import React, { useEffect, useState } from 'react';
+import { DialogBody } from '../DialogBody';
+import TextField from '@mui/material/TextField';
+import { FormattedMessage, useIntl } from 'react-intl';
+import { useDispatch } from 'react-redux';
+import { fetchRenameAssetDependants, updateRenameAssetDialog } from '../../state/actions/dialogs';
+import { cleanupAssetName, editorDisplay, getName, isEditableAsset } from '../../utils/content';
+import useActiveSiteId from '../../hooks/useActiveSiteId';
+import { DependenciesList } from '../DependenciesDialog';
+import useItemsByPath from '../../hooks/useItemsByPath';
+import { getParentPath } from '../../utils/path';
+import { UNDEFINED } from '../../utils/constants';
+import { isBlank } from '../../utils/string';
+import SecondaryButton from '../SecondaryButton';
+import { DialogFooter } from '../DialogFooter';
+import PrimaryButton from '../PrimaryButton';
+import { validateActionPolicy } from '../../services/sites';
+import { ConfirmDialog } from '../ConfirmDialog';
+import { renameContent } from '../../services/content';
+import { showErrorDialog } from '../../state/reducers/dialogs/error';
+import { translations } from '../CreateFileDialog/translations';
+import Typography from '@mui/material/Typography';
+import FormControlLabel from '@mui/material/FormControlLabel';
+import Checkbox from '@mui/material/Checkbox';
+import useSpreadState from '../../hooks/useSpreadState';
+import { DetailedItem } from '../../models';
+import Menu from '@mui/material/Menu';
+import MenuItem from '@mui/material/MenuItem';
+import useSelection from '../../hooks/useSelection';
+import Alert from '@mui/material/Alert';
+
+export function RenameAssetDialogContainer(props: RenameAssetContainerProps) {
+  const { onClose, onRenamed, path, value = '', allowBraces = false, type, dependantItems } = props;
```

Comment:
- OnClose is sent by the EnhancedDialog
- Keep dependencies and fetching all in local state